packages/eslint-plugin/src/rules/strict-boolean-expressions.ts

```diff
@@ -113,6 +114,9 @@ export default createRule<Options, MessageId>({
       conditionErrorNullableObject:
         'Unexpected nullable object value in conditional. ' +
         'An explicit null check is required.',
+      conditionErrorNullableObjectConvertToNullishCoalescing:
+        'Unexpected nullable object value in conditional with `||` or `&&`. ' +
+        'You should probably be using nullish coalescing instead.',
```

Comment:
If we move to this approach for objects, I see no reason to be inconsistent for the other types this rule works with: strings and numbers.

Currently, for both strings and numbers, the rule reports `conditionErrorNullableString` and `conditionErrorNullableNumber` respectively. These reports include both `value != null` and `value ?? 0` suggestions. So it's up to user to decide which suggestion should be applied in a particular situation. 

My take here is: what if we just add new suggestion with nullish coalescing to the existing `conditionErrorNullableObject` error instead of reporting new type of error without suggestions?

I see a several pros of this approach:

1) There are various cases where the logical expression can be used:

- Truth-testing context: `if (func || (() => null))`, ...
  Personally, I'd prefer to explicitly test for truth here (true/false), so if editor suggests me option of fixing to `!= null` and fixing to `?? ...`, I'd choose the former.

- Result-sensitive context: as in the original issue, where we care about the actual value of the expression result: `const foo = func || (() => null)`
  As mentioned in the original issue, fixing this to `!= null` will change the actual logic. So if editor suggests me to fix to `!= null` and fix to `?? ...`, I'd choose the latter.

  So unless rule can automatically determine which of the contexts mentioned above it operates in, I think it should provide both suggestions and leave the final decision up to the rule user.

2) If we decide to report error without providing suggestions for objects,  we should be consistent and do the same for numbers and strings (they actually provide the same `!= null` fix). I'm not sure this would be desirable behavior for all users.

What do you think?

Reply:
I think I agree with you ! But this is a bit more work, I'll try to find the time to do it.